go/service/main.go

```diff
@@ -731,9 +732,30 @@ func (d *Service) lockPIDFile() (err error) {
 }
 
 func (d *Service) ConfigRPCServer() (l net.Listener, err error) {
-	if l, err = d.G().BindToSocket(); err != nil {
+	// First, check to see if we've been launched with socket activation by
+	// systemd. If so, the socket is already open.
+	activationListeners, err := activation.Listeners(true)
+	if err != nil {
```

Comment:
I'm mainly worried about error'ing out here on let's say Windows or iOS

Reply:
The protocol here is that it just looks for specific environment variables, so I was _hoping_ we'd get away with this. Unfortunately it looks like the https://github.com/coreos/go-systemd project doesn't even build on Windows, so either way it looks like I'll be splitting out a Linux-specific file for this :p